velox/docs/functions/window.rst

```diff
@@ -0,0 +1,83 @@
+================
+Window functions
+================
+
+Velox window functions can be used to compute SQL window functions.
+
+Understanding the window function definition
+--------------------------------------------
+Window functions operate over all the input rows.
+
+Each window function can be thought to operate with an OVER clause
+that specifies how it is evaluated:
+
+.. code-block::
+
+   function(args) OVER (
+        [PARTITION BY expression]
+        [ORDER BY expression [ASC|DESC]]
+        [frame]
+   )
+
+A frame is one of:
+
+.. code-block::
+
+   {RANGE|ROWS} frame_start
+   {RANGE|ROWS} BETWEEN frame_start AND frame_end
+
+frame_start and frame_end can be any of:
+
+.. code-block::
+
+   UNBOUNDED PRECEDING
+   expression PRECEDING  -- only allowed in ROWS mode
+   CURRENT ROW
+   expression FOLLOWING  -- only allowed in ROWS mode
+   UNBOUNDED FOLLOWING
+
+More details:
+
+* The PARTITION BY fields separates the input rows into different partitions.
+
+  This is analogous to how the aggregate functions input is separated into different groups for evaluation.
+  If PARTITION BY fields are not specified, the entire input is treated as a single partition.
+
+* The ORDER BY fields determines the order in which input rows will be processed by the window function.
+
+  If ORDER BY fields are not specified, the ordering is undefined.
+
+* The frame clause specifies the sliding window of rows to be processed by the function for a given input row.
+
+  A frame can be ROWS type or RANGE type, and it runs from frame_start to frame_end.
+  If frame_end is not specified, a default value of CURRENT ROW is used.
+
+  In ROWS mode, CURRENT ROW refers specifically to the current row.
+
+  In RANGE mode, CURRENT ROW refers to any peer row of the current row for the purpose of the ORDER BY.
```

Comment:
I don't believe "peer row" term has been defined yet. Would you introduce it somewhere before this statement?

Reply:
Defined peer row just after this sentence.